### DISCUSSION TITLE
查找高亮如何设置

### DISCUSSION THREAD

**Discussion by uxfion** [nshen/learn-neovim-lua#156] [Q&A] [answered]
使用`/`进行查找时，只有在刚输入`/apt`的时候会高亮，在按下`enter`后，使用`n`或者`N`进行搜索的时候就没有高亮，有时候都不知道光标在哪。
针对这个问题，请问大家改如何解决？
<img width="432" alt="image" src="https://user-images.githubusercontent.com/44778029/209456316-dcef1801-4f57-40df-85b3-c957a403ccb5.png">
<img width="378" alt="image" src="https://user-images.githubusercontent.com/44778029/209456324-b227271f-2829-4112-8fbd-ce9b94286221.png">

**Comment by nshen** [comment]
insisvim好像没这个问题，不记得有改过哪里，你是用v2分支吗

https://user-images.githubusercontent.com/181506/209494465-64d4be7e-1e37-4060-9498-235f6266b514.mp4

**Reply by uxfion** [reply]
我现在用的是自己按照掘金小册上一步一步配的，但是发现查找有点问题，我不太清楚该怎么配置像您一样的高亮，我就只有在刚输入`/`的时候会橙色高亮，然后摁下`n`后橙色就没有了，也没有所有匹配字符的蓝底

**Comment by uxfion** [comment]
我对比了[InsisVim/lua/insis/basic.lua](https://github.com/nshen/InsisVim/blob/main/lua/insis/basic.lua)，发现关于搜索的基础配置都一样啊

```lua
-- 搜索大小写不敏感，除非包含大写
vim.o.ignorecase = true
vim.o.smartcase = true

-- 搜索不要高亮
vim.o.hlsearch = false

-- 边输入边搜索
vim.o.incsearch = true
```



请问在哪里设置搜索高亮？

**Comment by nshen** [comment, marked as answer]
可能是哪个组件覆盖了这些值吧，把这段加上试试呢？其他没有印象哪里修改过了
https://github.com/nshen/InsisVim/blob/990ba1c89effe98753ace29d70de17f2edf216c5/lua/insis/autocmds.lua#L66-L71

**Reply by uxfion** [reply]
添加了你上面说的就好了，感谢
